src/components/Calculator.js

```diff
@@ -1,36 +1,74 @@
-/* eslint-disable react/prefer-stateless-function */
-/* eslint-disable no-unused-vars */
-
 import { PureComponent } from 'react';
+import calculate from './logic/calculate';
 
 class Calculator extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      next: null,
+      operation: null,
+      total: null,
+    };
+  }
+
+  btnClick = (e) => {
+    this.setState((prev) => calculate(prev, e.target.innerText));
+  };
+
   render() {
```

Comment:
Great design @tarek102 🤝

- [ ] [Optional] Kindly implement all your functions and methods across all files using ES6 syntax